core/components/atoms/_chip/index.tsx

```diff
@@ -26,6 +26,7 @@ export const GenericChip = (props: GenericChipProps) => {
     classNames({
       ['Chip-icon']: true,
       [`Chip-icon--${align}`]: align,
+      [`Chip-icon-disabled--${align}`]: align === 'right' && disabled,
```

Comment:
Please change the className to be more specific to right icon. There is no need to give className with `${align}` condition when this is only applicable for right icon